app/lib/exchange-rate/money.ts

```diff
@@ -0,0 +1,480 @@
+
+    this._data = {
+      amount: amountInMinUnit.round(minUnit.decimals, Big.roundDown),
+      currency: data.currency,
+      amountUnit: minUnit,
+      initialUnit: unit,
+    };
+
+    Object.freeze(this);
+    Object.freeze(this._data);
+  }
+
+  static create(props: MoneyInput): Money {
+    return new Money(props);
+  }
+
+  /**
+   * Sum an array of moneys.
+   *
+   * If the array is empty, a currency must be specified so that 0 can be returned in that currency.
+   *
+   * The precision, etc., is based on the first item in the array.
+   * If the array is empty, the options object will be used instead.
+   */
+  static sum(moneys: Money[], currency?: string): Money {
+    if (moneys.length === 0 && currency === undefined) {
+      throw new Error(
+        "Currency must be set when summing an empty list of money's",
+      );
+    }
+
+    if (moneys.length === 0 && currency !== undefined) {
+      return new Money({ amount: 0, currency });
+    }
+
+    return moneys.slice(1).reduce((sum, money) => sum.add(money), moneys[0]);
+  }
+
+  static max(moneys: Money[]): Money {
+    if (moneys.length === 0) {
+      throw new Error('Need at least one money for comparison');
+    }
+    return moneys.reduce(
+      (max, money) => (money.greaterThan(max) ? money : max),
+      moneys[0],
+    );
+  }
+
+  static min(moneys: Money[]): Money {
+    if (moneys.length === 0) {
+      throw new Error('Need at least one money for comparison');
+    }
+    return moneys.reduce(
+      (min, money) => (money.lessThan(min) ? money : min),
+      moneys[0],
+    );
+  }
+
+  /**
+   * Compare two money objects.
+   *
+   * 1 if money1 is greater than money2
+   * 0 if equal
+   * -1 if money1 is less than money2
+   *
+   * This can be plugged directly into array.sort(),
+   * and it will cause the array to be sorted in ascending order.
+   */
+  static compare(money1: Money, money2: Money): number {
+    money1.assertSameCurrency(money2);
+    return money1.amount().cmp(money2.amount());
+  }
+
+  get currency(): string {
+    return this._data.currency;
+  }
+
+  /**
+   * Returns the money amount in the requested currency unit. If no unit is provided it uses the
+   * default/base unit (bitcoin, dollar, ...).
+   *
+   * If the requested unit does not have enough precision the number will be rounded down to the number of decimals of
+   * the requested unit with round down. For example if the requested unit is 'bitcoin' and the money instance has 4567
+   * msat, the return value will be 0.00000004
+   *
+   * @param unit The unit of the currency to use for the amount. Default is base currency unit (bitcoin for BTC, dollar
+   * for USD, etc.)
+   */
+  amount = (unit?: string): Big => {
+    const unitToCalculate = this.getCurrencyUnit(unit);
+    const multiplier = this._data.amountUnit.factor.div(unitToCalculate.factor);
+    return this._data.amount
+      .mul(multiplier)
+      .round(unitToCalculate.decimals, Big.roundDown);
```

Comment:
why always round down? I guess its better to underestimate rather than overestimate, but should this be configurable?

Reply:
if you check the comment above this method it says:
```
If the requested unit does not have enough precision the number will be rounded down to the number of decimals of the requested unit. For example if the requested unit is 'btc' and the money instance has 4567 msat, the return value will be 0.00000004
```

if we weren't rounding down `4567 msat` would be rounded to 5 sats instead of 4. when there is a loss of precision like in this case I don't see a better way. otherwise we would be printing money :)